code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -557,3 +591,43 @@
  */
 /obj/machinery/atmospherics/proc/paint(paint_color)
 	return FALSE
+
+/**
+ * Disconnects all nodes from ourselves, remove us from the node's nodes.
+ * Nullify our parent pipenet
+ */
+/obj/machinery/atmospherics/proc/disconnect_nodes()
+	for(var/i in 1 to device_type)
+		var/obj/machinery/atmospherics/node = nodes[i]
+		if(node)
+			if(src in node.nodes) //Only if it's actually connected. On-pipe version would is one-sided.
+				node.disconnect(src)
+			nodes[i] = null
+		if(parents[i])
+			nullify_pipenet(parents[i])
+
+/**
+ * Connects all nodes to ourselves, add us to the node's nodes.
+ * Calls atmos_init() on the node and on us.
+ */
+/obj/machinery/atmospherics/proc/connect_nodes()
+	atmos_init()
+	for(var/i in 1 to device_type)
+		var/obj/machinery/atmospherics/node = nodes[i]
+		node = nodes[1]
+		if(node)
+			node.atmos_init()
+			node.add_member(src)
+	SSair.add_to_rebuild_queue(src)
+
+/**
+ * Easy way to toggle nodes connection and disconnection.
+ *
+ * Arguments:
+ * * disconnect - if TRUE, disconnects all nodes. If FALSE, connects all nodes.
+ */
+/obj/machinery/atmospherics/proc/change_nodes_connection(disconnect)
+	if(disconnect)
+		disconnect_nodes()
+		return
+	connect_nodes()
```

Comment:
This seems like kinda strange behavior to stick on base atmospherics, how common is this pattern?

Reply:
is used on several machines, like the thermomachine and the crystallizer. tho i will move this to the base_component instead of base atmosmachinery